app/models/reservation.rb

```diff
@@ -5,13 +5,13 @@ class Reservation < ApplicationRecord # :nodoc:
   enum status: %i[pending confirmed canceled]
 
   # Associations
-  belongs_to :room
   belongs_to :guest
+  belongs_to :room
   belongs_to :user
 
   # Validations
-  validates :room_id, presence: true
-  validates :guest_id, presence: true
+  validates :status, inclusion: { in: statuses.keys }
   validates :checkin_at, presence: true
   validates :checkout_at, presence: true
+  validates :book_code, presence: true, uniqueness: true
```

Comment:
What is this? Maybe in the future, let's avoid adding additional things that are not part of the ticket, or add to the ticket description to be more clearer to the reviewers.

Reply:
You're right, will keep the next PR's shorter, I wanted to fix at maximum the validations, factories and specs, and the book code validation should be tested as well.